corehq/apps/repeaters/models.py

```diff
@@ -535,13 +536,19 @@ def fire(self, force_send=False):
     def post(self, post_info, tries=0):
         tries += 1
         try:
-            response = simple_post_with_logged_timeout(
-                self.domain,
-                post_info.payload,
-                self.url,
-                headers=post_info.headers,
-                timeout=POST_TIMEOUT,
-            )
+            generator = self.get_payload_generator(self.format_or_default_format())
+            if generator.format_label == 'XML' and hasattr(self.repeater, 'operation'):
+                client = Client(self.url)
+                response = client.service[self.repeater.operation](**post_info.payload)
+                return self.handle_success(response)
```

Comment:
Can you explicitly move the `self.handle_response` for the POST out of the `else` clause so that the code flow is easier to follow? Also, I'd added a comment alongside this line so that it was more obvious why we were calling `handle_success` instead of letting this class handle the response.